pkg/network/deviceinfo/deviceinfo_test.go

```diff
@@ -0,0 +1,165 @@
+			networks := []virtv1.Network{*libvmi.MultusNetwork("foo", "default/nad1")}
+			interfaces := []virtv1.Interface{newBindingPluginInterface("foo", deviceInfoPlugin)}
+			_, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).To(HaveOccurred())
+		},
+		Entry("when networkStatus annotation is empty", ""),
+		Entry("when networkStatus annotation has invalid format", "invalid"),
+	)
+
+	DescribeTable("should prepare empty network device info annotation",
+		func(networkList []virtv1.Network, interfaceList []virtv1.Interface, networkStatusAnnotationValue string) {
+			networkDeviceInfoMap, err :=
+				deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networkList, interfaceList, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(networkDeviceInfoMap).To(BeEmpty())
+		},
+		Entry("when there is no interface with device info binding plugin",
+			[]virtv1.Network{
+				*virtv1.DefaultPodNetwork(),
+				*libvmi.MultusNetwork("foo", "default/nad1"),
+			},
+			[]virtv1.Interface{
+				libvmi.InterfaceDeviceWithMasqueradeBinding(),
+				newBindingPluginInterface("foo", nonDeviceInfoPlugin),
+			},
+			fmt.Sprintf(networkStatusWithMixedNetworksFmt, booHashedIfaceName, fooHashedIfaceName),
+		),
+		Entry("when the interface is not in the multus status",
+			[]virtv1.Network{*libvmi.MultusNetwork("notfoo", "default/nad1")},
+			[]virtv1.Interface{newBindingPluginInterface("notfoo", deviceInfoPlugin)},
+			networkStatusWithPrimaryInterfaceOnly,
+		),
+	)
+
+	DescribeTable("should prepare network device info annotation"+
+		" when given Interfaces{1X primary- no device info, 1X no device info, 1X with device info, 1X not in multus status}",
+		func(booPodIfaceName, fooPodIfaceName string) {
+			networks := []virtv1.Network{
+				*virtv1.DefaultPodNetwork(),
+				*libvmi.MultusNetwork("boo", "default/nad1"),
+				*libvmi.MultusNetwork("foo", "default/nad2"),
+				*libvmi.MultusNetwork("doo", "default/nad3"),
+			}
+			interfaces := []virtv1.Interface{
+				libvmi.InterfaceDeviceWithMasqueradeBinding(),
+				libvmi.InterfaceDeviceWithBridgeBinding("boo"),
+				newBindingPluginInterface("foo", deviceInfoPlugin),
+				newBindingPluginInterface("doo", deviceInfoPlugin),
+			}
+			networkStatusAnnotationValue := fmt.Sprintf(networkStatusWithMixedNetworksFmt, booPodIfaceName, fooPodIfaceName)
+			expectedMap :=
+				map[string]*networkv1.DeviceInfo{"foo": {Type: "pci", Version: "1.0.0", Pci: &networkv1.PciDevice{PciAddress: "0000:65:00.2"}}}
+			actualMap, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, networkStatusAnnotationValue, bindingPlugins)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actualMap).To(Equal(expectedMap))
+		},
+
+		Entry("with hashed pod interface names",
+			booHashedIfaceName, fooHashedIfaceName,
+		),
+		Entry("with ordinal pod interface names",
+			"net1", "net2"),
```

Comment:
Can such a scenario exist?
We had network binding networks when the old naming was used?

Reply:
No, we switched the naming at version v1.0.0 [1]  , network binding plugin introduced at version v1.1.0 [2]

[1] https://github.com/kubevirt/kubevirt/pull/9268
[2] https://github.com/kubevirt/kubevirt/pull/10284

Reply:
done.